utils/cache.go

```diff
@@ -0,0 +1,79 @@
+package utils
+
+import (
+	"reflect"
+	"sync"
+)
+
+type Cache[T any] interface {
+	Insert(int, T)
+	Query(int) (T, bool)
+}
+
+var (
+	DefaultDoubleCacheCapacity = 1024 * 1024
```

Comment:
[question]
where does this value come from? 
[nit][not blocking]
mind adding a doc comment?

Reply:
Just a random value. 1 MB of default cache?